src/core/public/application/scoped_history.ts

```diff
@@ -111,6 +115,25 @@ export class ScopedHistory<HistoryLocationState = unknown>
     return this.parentHistory.action;
   }
 
+  /**
+   * Fetches the parent location state
+   */
+  public fetchLocationState<T extends unknown = unknown>(): T {
+    this.verifyActive();
+    return cloneDeep(this.parentLocationState) as T;
+  }
```

Comment:
Why is this necessary? The state should already be available via `ScopedHistory#location` getter. If not, I think there is a bug somewhere that isn't forwarding the state.

Reply:
When I first began working on this, I noticed that `ScopedHistory.location.state` was always undefined in Lens, even if there was content in the `parentHistory.location.state`. That was why I added this workaround. Now for the life of me, I can't reproduce that original problem, so this workaround has been removed.